netbox/dcim/filtersets.py

```diff
@@ -1400,6 +1400,12 @@ class DeviceComponentFilterSet(django_filters.FilterSet):
         to_field_name='slug',
         label=_('Device role (slug)'),
     )
+    device_status = django_filters.ModelMultipleChoiceFilter(
```

Comment:
I believe this should be a (non-model) MultipleChoiceFilter, like [DeviceFilterSet.status](https://github.com/netbox-community/netbox/blob/cc72a58c1e92b2e3948ff02ca581bbc510935940/netbox/dcim/filtersets.py#L1041).

Reply:
I thought the same, but received an error when switching to a MultipleChoiceFilter. Below is the traceback:

```
(.venv) parallels@ubuntu-linux-22-04-02-desktop:~/code/netbox/netbox$ python manage.py runserver
Watching for file changes with StatReloader
Performing system checks...

Traceback (most recent call last):
  File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/home/parallels/code/netbox/.venv/lib/python3.10/site-packages/django/utils/autoreload.py", line 64, in wrapper
    fn(*args, **kwargs)
  File "/home/parallels/code/netbox/.venv/lib/python3.10/site-packages/django/core/management/commands/runserver.py", line 133, in inner_run
    self.check(display_num_errors=True)
  File "/home/parallels/code/netbox/.venv/lib/python3.10/site-packages/django/core/management/base.py", line 486, in check
    all_issues = checks.run_checks(
  File "/home/parallels/code/netbox/.venv/lib/python3.10/site-packages/django/core/checks/registry.py", line 88, in run_checks
    new_errors = check(app_configs=app_configs, databases=databases)
  File "/home/parallels/code/netbox/.venv/lib/python3.10/site-packages/django/core/checks/urls.py", line 14, in check_url_config
    return check_resolver(resolver)
  File "/home/parallels/code/netbox/.venv/lib/python3.10/site-packages/django/core/checks/urls.py", line 24, in check_resolver
    return check_method()
  File "/home/parallels/code/netbox/.venv/lib/python3.10/site-packages/django/urls/resolvers.py", line 519, in check
    for pattern in self.url_patterns:
  File "/home/parallels/code/netbox/.venv/lib/python3.10/site-packages/django/utils/functional.py", line 47, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/home/parallels/code/netbox/.venv/lib/python3.10/site-packages/django/urls/resolvers.py", line 738, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
  File "/home/parallels/code/netbox/.venv/lib/python3.10/site-packages/django/utils/functional.py", line 47, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/home/parallels/code/netbox/.venv/lib/python3.10/site-packages/django/urls/resolvers.py", line 731, in urlconf_module
    return import_module(self.urlconf_name)
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/parallels/code/netbox/netbox/netbox/urls.py", line 10, in <module>
    from netbox.graphql.schema import schema
  File "/home/parallels/code/netbox/netbox/netbox/graphql/schema.py", line 5, in <module>
    from circuits.graphql.schema import CircuitsQuery
  File "/home/parallels/code/netbox/netbox/circuits/graphql/schema.py", line 7, in <module>
    from .types import *
  File "/home/parallels/code/netbox/netbox/circuits/graphql/types.py", line 11, in <module>
    from .filters import *
  File "/home/parallels/code/netbox/netbox/circuits/graphql/filters.py", line 3, in <module>
    from circuits import filtersets, models
  File "/home/parallels/code/netbox/netbox/circuits/filtersets.py", line 5, in <module>
    from dcim.filtersets import CabledObjectFilterSet
  File "/home/parallels/code/netbox/netbox/dcim/filtersets.py", line 1479, in <module>
    class ConsolePortFilterSet(
  File "/home/parallels/code/netbox/.venv/lib/python3.10/site-packages/django_filters/filterset.py", line 63, in __new__
    new_class.base_filters = new_class.get_filters()
  File "/home/parallels/code/netbox/netbox/netbox/filtersets.py", line 222, in get_filters
    additional_filters.update(cls.get_additional_lookups(existing_filter_name, existing_filter))
  File "/home/parallels/code/netbox/netbox/netbox/filtersets.py", line 179, in get_additional_lookups
    resolve_field(field, lookup_expr)  # Will raise FieldLookupError if the lookup is invalid
  File "/home/parallels/code/netbox/.venv/lib/python3.10/site-packages/django_filters/utils.py", line 204, in resolve_field
    query = model_field.model._default_manager.all().query
AttributeError: 'NoneType' object has no attribute 'model'
```

Looking at the rest of the filters for DeviceComponentFilterSet, it appears they're using a ModelMultipleChoiceFilter because a queryset is required for the DeviceComponentFilterSet? Sorry, I'm still trying to understand django-filters.